src/components/Button/Button.scss

```diff
@@ -149,6 +149,13 @@ $spinner-size: rem(20px);
     margin-right: 0;
   }
 
+  &.destructive {
```

Comment:
@sarahill What do you think color-wise here? Should destructive card actions should match destructive action list items in being dark red? The reason action list item content is dark red is because there wouldn't be enough contrast between the focus/hover background colors, but plain buttons don't have background so could arguably work as is here.

Reply:
<img width="992" alt="Screen Shot 2019-05-31 at 11 36 55 AM" src="https://user-images.githubusercontent.com/18447883/58717094-71431480-8398-11e9-976c-63b36c8abb36.png">

Reply:
The 2 different reds feel odd. I'm good with using `red dark` for all of these as far as color goes.

Reply:
Should we have a hover/focus state similar to a regular plain button that darkens the content? 

Current

```scss
  &.destructive {
    color: color('red');
    &:hover {
      color: color('red', 'dark');
    }
  }
```

![2019-05-31 12 20 15](https://user-images.githubusercontent.com/18447883/58719831-9dfa2a80-839e-11e9-8443-a52852867343.gif)

---

Dark red to darker red might be too drastic of a transition 🤔 

```scss
  &.destructive {
    color: color('red', 'dark');
    &:hover {
      color: color('red', 'darker');
    }
  }
```

![2019-05-31 12 22 48](https://user-images.githubusercontent.com/18447883/58719981-f4ffff80-839e-11e9-9ca0-f8cb9f108bc6.gif)

---

Maybe something like

```scss
  &.destructive {
    color: color('red', 'dark');
    &:hover {
      color: darken(color('red', 'dark'), 10%);
    }
  }
```

![2019-05-31 12 22 48](https://user-images.githubusercontent.com/18447883/58720151-5cb64a80-839f-11e9-9025-7f5ab183e612.gif)

Reply:
Ah I didn't see you commented on this. Related to my comment here https://github.com/Shopify/polaris-react/pull/1603#issuecomment-497770381

I'm mixed on modifying the red darker. I agree it's pretty drastic but it's also another example of where our color system starts to fall short (lack of range). I'm ok with this approach but something we should keep in mind / consider with the color system updates.

Reply:
💯

Reply:
To clarify, we should go with the third variation above?

Reply:
> I'm mixed on modifying the red darker

Understandable, though I just did a quick search and it doesn't look like `color('red', 'darker')` is used anywhere in the system or in `web`, so it would likely be low risk as far as UI regression is concerned.